e2e-polybft/framework/test-cluster.go

```diff
@@ -565,18 +593,31 @@ func NewTestCluster(t *testing.T, validatorsCount int, opts ...ClusterOption) *T
 			args = append(args, "--native-token-config", cluster.Config.NativeTokenConfigRaw)
 		}
 
-		for _, premine := range cluster.Config.Premine {
-			args = append(args, "--premine", premine)
+		tokenConfig, err := polybft.ParseRawTokenConfig(cluster.Config.NativeTokenConfigRaw)
+		require.NoError(t, err)
+
+		if len(cluster.Config.Premine) != 0 && tokenConfig.IsMintable {
+			// only add premine flags in genesis if token is mintable
+			for _, premine := range cluster.Config.Premine {
+				args = append(args, "--premine", premine)
+			}
+		}
+
+		burnContract := cluster.Config.BurnContract
+		if burnContract != nil {
+			args = append(args, "--burn-contract",
+				fmt.Sprintf("%d:%s:%s",
+					burnContract.BlockNumber, burnContract.Address, burnContract.DestinationAddress))
 		}
 
-		if len(cluster.Config.StakeAmounts) > 0 {
+		if tokenConfig.IsMintable && len(cluster.Config.StakeAmounts) != 0 {
```

Comment:
We should allow providing stake in case stake token is non-native token (and native token is from an L1).

Also probably we should introduce such e2e test in the future.

Reply:
Let's address this through a separate PR.